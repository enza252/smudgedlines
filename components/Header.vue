<template>
  <v-app-bar
    color="primary"
    style="border-bottom: 1px solid #ffffff !important;"
  >
    <v-app-bar-nav-icon>
      <template v-slot:default>
        <v-img :src="slLogoPath" max-height="50" max-width="50"/>
      </template>
    </v-app-bar-nav-icon>

    <v-app-bar-title>
      <template v-slot:default>
        <h3 style="color: white">Smudged Lines</h3>
      </template>
    </v-app-bar-title>
    <v-spacer></v-spacer>
    <div v-for="(button, index) in navBarButtons" :key="index">
      <v-tooltip bottom>
        <template v-slot:activator="{ on, attrs }">
          <v-btn color="white" class="ma-1" text v-on="on" v-bind="attrs" @click="$vuetify.goTo(`#${button.id}-header`)">
            {{ button.buttonText }}
            <v-icon right >{{ button.icon }}</v-icon>
          </v-btn>
        </template>
        <span>{{ button.buttonText }}</span>
      </v-tooltip>
    </div>
  </v-app-bar>
</template>

<script>
import { defineComponent } from '@nuxtjs/composition-api'

import { mdiTshirtCrewOutline, mdiInformationOutline, mdiHeartOutline, mdiCartOutline } from '@mdi/js'
export default defineComponent({
  name: 'Header',
  data: () => ({
    navBarButtons: [
      {
        icon: mdiTshirtCrewOutline,
        buttonText: 'Designs',
        id: 'designs'
      },
      {
        icon: mdiHeartOutline,
        buttonText: 'Care',
        id: 'care'
      },
      {
        icon: mdiInformationOutline,
        buttonText: 'About',
        id: 'about'
      },
      {
        icon: mdiCartOutline,
        buttonText: 'Store',
        id: 'store'
      }
    ],
    slLogoPath: '/logo/sl-icon-white-200x200.png'
  }),
})
</script>
