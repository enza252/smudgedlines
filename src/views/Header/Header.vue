<template>
  <div>
    <v-app-bar
      color="green"
    >
      <v-app-bar-nav-icon></v-app-bar-nav-icon>

      <v-toolbar-title>Smudged Lines</v-toolbar-title>
      <v-spacer></v-spacer>

      <v-btn>
        Designs
        <v-icon right >mdi-home</v-icon>

      </v-btn>
      <v-btn>
        About
        <v-icon right>
          mdi-t-shirt-crew-outline
        </v-icon>
      </v-btn>
    </v-app-bar>
  </div>
</template>

<script>
export default {
  name: 'Header'
}
</script>
