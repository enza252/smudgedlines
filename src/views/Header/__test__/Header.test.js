import { render, screen } from '@testing-library/vue'
import Vuetify from 'vuetify'
import Header from '@/views/Header/Header'

const options = {
  routes: [],
  vuetify: new Vuetify()
}
describe('Tests the Header view is rendered correctly', () => {
  it('Renders the Header and checks against a snapshot', async () => {
    const { asFragment } = await render(Header, options)
    expect(asFragment).toMatchSnapshot()
  })
})
