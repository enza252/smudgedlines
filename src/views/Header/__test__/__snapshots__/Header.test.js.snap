// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Header view is rendered correctly Renders the Header and checks against a snapshot 1`] = `undefined`;
